Add ERC: In-Ground Asset Token

solomonashok opened the pull request:
ERC-IGA proposes a standardized, permissioned, compliance-aligned token model for representing in-ground geological assets on Ethereum.
It introduces domain-specific metadata primitives for:

geological provenance (JORC / NI 43-101 / SAMREC / PERC)

Competent Person (CP/QP) signatures

reserve classification

dynamic extraction/depletion state

ESG and rehabilitation bonding

jurisdiction-aware and identity-restricted transfers

This ERC enables miners, streaming/royalty companies, commodity funds, banks, and sovereign entities to tokenize verified, regulator-aligned mineral reserves in a safe and interoperable manner.

solomonashok committed:
Add ERC-IGA: In-Ground Asset Tokenization Standard

solomonashok committed:
Updated ERC-IGA: In-Ground Asset Tokenization Standard

solomonashok committed:
Update-2 ERC-IGA: In-Ground Asset Tokenization Standard

abcoathup commented on ERCS/erc-iga.md:
```diff
@@ -0,0 +1,170 @@
+---
+eip: <to be assigned>
```
```suggestion
eip: 8097
```
Assigning next sequential EIP/ERC/RIP number.
Numbers are assigned by editors & associates.

Please also update the filename.

abcoathup commented on ERCS/erc-iga.md:
```diff
@@ -0,0 +1,170 @@
+eip: <to be assigned>
+title: ERC-IGA In-Ground Asset Tokenization Standard
+author: Solomon Ashok J N <solomon@ingroundassets.com>
+discussions-to: https://ethereum-magicians.org/c/standards/erc/
```
Please create a discussions topic in Eth Magicians with a link to this PR.

solomonashok committed:
Update EIP-8097 draft, rename file, apply title change, and align wording with EIP guidelines

- Updated header to `eip: 8097` and `title: In-Ground Asset Token`
- Renamed file to ERCS/eip-8097.md
- Replaced references to “ERC-IGA” with neutral phrasing
- Added clarification that the ERC is in Draft status
- Adjusted content to remove branding/marketing language
- Added notes on planned updates (metadata examples, security considerations, reference implementation)

solomonashok committed:
Merge branch 'master' into erc-iga

solomonashok committed:
Merge branch 'master' into erc-iga

SamWilsn commented on ERCS/erc-8097.md:
```diff
@@ -0,0 +1,170 @@
+
+## Simple Summary
+
+ERC-IGA defines a standard interface and lifecycle model for tokenizing verified **in-ground geological assets** on Ethereum. It introduces primitives for geological provenance, reserve classification, dynamic extraction-linked supply, ESG and rehabilitation bonding, and regulated transfer control. ERC-IGA acts as the compliance and verification framework for representing *real-world mineral rese ...
```
> [...] ESG [...]

Please expand uncommon abbreviations the first time they are used.

SamWilsn commented on ERCS/erc-8097.md:
```diff
@@ -0,0 +1,170 @@
+- **IGA‑T** — Fixed‑supply in‑ground reserve tokens  
+- **IGA‑X** — Extraction-linked, dynamically adjusting supply  
+- **IGA‑C** — ESG, credit, rehabilitation, or structured note modules  
+
```
This is a bit too light, even for a draft. I need to be able to determine how feasible this proposal is, and I cannot from this as is.

SamWilsn commented on ERCS/erc-8097.md:
```diff
@@ -0,0 +1,170 @@
+- **ERC‑721** for single‑asset geological titles  
+- **ERC‑1155** for multi‑commodity deposits  
+- **ERC‑3643** for identity‑gated regulatory compliance  
+
```
Don't bold ERC/EIP references

SamWilsn commented on ERCS/erc-8097.md:
```diff
@@ -0,0 +1,170 @@
+- **Oracle Manipulation**: extraction data MUST use multi‑sig, CP, or decentralized oracle networks.  
+- **Over‑Issuance Risk**: IGA‑X MUST enforce remainingInGround >= totalSupply.  
+- **Jurisdiction Controls**: transfers MUST check on-chain regional restrictions.  
+- **ESG Enforcement**: rehabilitation bonding MUST be unfreezable until obligations are met.  
```
Keep requirements (defined with UPPERCASE keywords) in the specification section, but definitely do discuss their implications in this section.

solomonashok committed:
chore: align ERC-8097 with EIP-1 terminology and reviewer feedback

solomonashok committed:
Merge branch 'master' into erc-iga

solomonashok committed:
chore: align ERC-8097 with EIP-1 terminology and reviewer feedback-2

solomonashok committed:
chore: align ERC-8097 with EIP-1 terminology and reviewer feedback-3

solomonashok committed:
chore: align ERC-8097 with EIP-1 terminology and reviewer feedback-4

solomonashok committed:
align ERC-8097 with EIP-1 terminology and reviewer feedback-5

solomonashok commented:
Thank you for the helpful feedback — we’ve updated the proposal accordingly.

Added a dedicated “Terminology & External Standards” section after the Abstract, with authoritative links and clarification that these standards are referenced inputs, not redefined.

Expanded the Specification to clarify feasibility by explicitly defining:

On-chain vs off-chain data boundaries

Mandatory hash anchoring and versioning

Clear MUST / MAY constraints aligned with EIP conventions.

Reorganized structure so:

Motivation contains problem framing and ecosystem justification.

Rationale now strictly explains design decisions.

Moved all normative requirements into the Specification and used Security Considerations to explain their implications.

Updated naming to use “In-Ground Asset Tokenization Standard” as the friendly name and reserve ERC-8097 strictly as the proposal identifier.

We believe these changes improve clarity, feasibility assessment, and alignment with EIP standards. Thank you again for the guidance.

solomonashok committed:
Merge branch 'master' into erc-iga

solomonashok committed:
Merge branch 'master' into erc-iga

SamWilsn commented on ERCS/erc-8097.md:
```diff
@@ -0,0 +1,170 @@
+
+Unlike traditional fungible or non-fungible token standards, in-ground assets require:
+
+- Geological metadata structures derived from technical reports (JORC, NI 43‑101, SAMREC, PERC, S‑K 1300)
```
https://github.com/ethereum/ERCs/pull/1387#discussion_r2710211891

SamWilsn commented on ERCS/erc-8097.md:
```diff
@@ -0,0 +1,170 @@
+- Immutable resource statement anchoring with cryptographic versioning
+- Extraction-linked state updates
+- Jurisdictional and identity-gated transfers
+- ESG, rehabilitation, and bonding primitives
```
https://github.com/ethereum/ERCs/pull/1387#discussion_r2710207503

SamWilsn commented on ERCS/erc-8097.md:
```diff
@@ -0,0 +1,170 @@
+
+## Rationale
+
+In-ground assets require depletion-aware, compliance-first modeling not addressed by existing token standards. This specification defines a minimal yet extensible framework to support institutional-grade mineral asset representation.
```
This is just justifying the entire proposal, not specific design decisions you made.

SamWilsn commented on ERCS/erc-8097.md:
```diff
@@ -0,0 +1,170 @@
+- **IEXO (Extraction & Depletion Object)**
+  - MUST track monotonically decreasing `remainingInGround`
+  - MUST revert updates that exceed initial reserves
+  - MUST emit depletion events
```
There still isn't enough technical meat here. What does "signed" mean? What is a "depletion event"? What is a "jurisdiction identifier"?

SamWilsn commented on ERCS/erc-8097.md:
```diff
@@ -0,0 +1,170 @@
+
+- **IGA-T** — Fixed-supply in-ground reserve tokens
+- **IGA-X** — Extraction-linked dynamic supply
+- **IGA-C** — ESG, rehabilitation, or credit modules
```
Are these going to be defined in this proposal or in other ones?

SamWilsn commented on ERCS/erc-8097.md:
```diff
@@ -0,0 +1,170 @@
+- NI 43-101 – Canadian National Instrument 43-101
+- SAMREC – South African Mineral Reporting Codes
+- PERC – Pan-European Reserves and Resources Reporting Committee
+- SEC S-K 1300 – United States SEC mining disclosure standard
```
Please follow the guidelines in EIP-1 and EIP-5757 for external links.

solomonashok committed:
technical definitions

solomonashok committed:
Merge branch 'ethereum:master' into erc-iga

solomonashok committed:
Create assets/erc-8097/contracts/ERC8097.sol

solomonashok committed:
created assets/erc-8097/contracts/IERC8097.sol

solomonashok committed:
created assets/erc-8097/test/ERC8097.t.sol

solomonashok committed:
created foundry, readme files

solomonashok committed:
Updated erc-8097.md tech

solomonashok committed:
Update erc-8097.md

solomonashok committed:
Rename ERC8097.sol to ERC-8097.sol

solomonashok committed:
Rename IERC8097.sol to IERC-8097.sol

solomonashok committed:
Rename ERC8097.t.sol to ERC-8097.t.sol

solomonashok committed:
Update erc-8097.md ref

solomonashok committed:
Update erc-8097.md ref fix

solomonashok committed:
Update erc-8097.md fix

solomonashok committed:
Update erc-8097.md ref path cahnge to eip

solomonashok committed:
Update erc-8097.md fix

solomonashok committed:
Merge branch 'master' into erc-iga

solomonashok committed:
Update erc-8097.md

solomonashok committed:
Update ERC-8097.sol

solomonashok committed:
Update IERC-8097.sol

solomonashok committed:
Update ERC-8097.t.sol

solomonashok committed:
Update README.md

solomonashok committed:
Update erc-8097.md

solomonashok committed:
Merge branch 'master' into erc-iga

solomonashok committed:
Merge branch 'master' into erc-iga
